napalm_brocade/brocade.py

```diff
@@ -0,0 +1,428 @@
+                break
+
+        # Initialize 'power', 'fan', and 'temperature' to default values
+        # (not implemented)
+
+        environment.setdefault('power', {})
+        environment['power']['invalid'] = {
+            'status': True,
+            'output': -1.0,
+            'capacity': -1.0
+            }
+        environment.setdefault('fans', {})
+        environment['fans']['invalid'] = {'status': True}
+        environment.setdefault('temperature', {})
+        environment['temperature']['invalid'] = {
+            'is_alert': False,
+            'is_critical': False,
+            'temperature': -1.0}
+        return environment
+
+    def get_facts(self):
+        cmd = "show system"
+        fact_table = {}
+        output = self.device.send_command(cmd)
+        output = output.split('\n')
+        output = output[3:-1]
+
+        for line in output:
+            if len(line) == 0:
+                return {}
+
+            match = re.search("Up Time.*: (.*)$", line)
+            if match:
+                uptime = match.group(1)
+                fact_table["uptime"] = uptime
+
+            match = re.search("^(.*) Version.*: (.*)$", line)
+            if match:
+                os_type = match.group(1)
+                os_version = match.group(2)
+                fact_table["os_type"] = os_type
+                fact_table["os_version"] = os_version
+
+            match = re.search("^Management IP.*: (.*)$", line)
+            if match:
+                mgmt_ip = match.group(1)
+                fact_table["management_ip"] = mgmt_ip
+
+            match = re.search("^Fan . is (.*)$", line)
+            if match:
+                fan_status = match.group(1)
+                fact_table["fan"] = fan_status
+
+        return fact_table
+
+    def get_vlan_table(self):
+        """
+        Get VLAN table.
+        """
+        vlan_table = []
+
+        vlan_cmd = 'show vlan brief'
+        output = self.device.send_command(vlan_cmd)
+        output = output.split('\n')
+
+        # Skip the first two lines which is the header
+        output = output[5:-1]
+
+        for line in output:
+            if len(line) == 0:
+                continue
+
+            vlans = line.split()
+            entry = {
+                'vlan': vlans[0],
+                'name': vlans[1]
+                }
+            vlan_table.append(entry)
+
+        return vlan_table
+
+    def get_arp_table(self):
+        """
+        Get ARP table.
+        """
+        arp_table = []
+
+        arp_cmd = 'show arp'
+        output = self.device.send_command(arp_cmd)
+        output = output.split('\n')
+
+        # Skip the first two lines which is the header
+        output = output[2:-1]
+
+        for line in output:
+            if len(line) == 0:
+                return {}
+            if len(line.split()) == 6:
+                address, mac, interface, macresolved, age, typ = line.split()
+                try:
```

Comment:
Not worth raising error for that - it may happen to not have the age of some entries.

Do instead: ```napalm_base.helpers.convert(age, 0.0)``` which will default to 0.0 when age is not valid.